Sources/Orbit/Components/HorizontalScroll.swift

```diff
@@ -23,301 +28,521 @@ public enum HorizontalScrollItemHeight {
+    ///
+    /// Scrolling can snap to leading edge by using `itemWidth` variants that support snapping.
+    ///
+    /// Can be scrolled programatically when wrapped inside `HorizontalScrollReader`.
+    /// This is currently limited only to non-snapping variant.
+    ///
     /// - Parameters:
     ///   - spacing: Spacing between items. Default value is `.small`.
     ///   - itemWidth: Horizontal sizing style of all items. Default value is `.ratio()`.
     ///   - itemHeight: Horizontal sizing style of all items. Default value is `.intrinsic`.
     ///   - maxItemWidth: Maximal item width that caps the result of `itemWidth`.
     ///                   Default value is derived from ``Layout/readableMaxWidth``.
-    ///   - minHeight: Minimal height of component. Default value is `nil`.
-    ///   - horizontalPadding: Horizontal padding inside the ScrollView. Default value is `.xSmall`.
-    ///   - verticalPadding: Horizontal padding inside the ScrollView. Can be used to fix overlay issues. Default value is `.xSmall`.
     ///   - content: Child items that will be horizontally laid out based on above parameters.
+    /// - Warning: when using snapping effect than programatic scrolling using `ScrollViewReader`
+    ///            is not possible.
     public init(
         spacing: CGFloat = .small,
         itemWidth: HorizontalScrollItemWidth = .ratio(),
         itemHeight: HorizontalScrollItemHeight = .intrinsic,
         maxItemWidth: CGFloat? = Layout.readableMaxWidth - 270,
-        minHeight: CGFloat? = nil,
-        horizontalPadding: CGFloat = .xSmall,
-        verticalPadding: CGFloat = .xSmall,
         @ViewBuilder content: () -> Content
     ) {
         self.spacing = spacing
         self.itemWidth = itemWidth
         self.itemHeight = itemHeight
         self.maxItemWidth = maxItemWidth
-        self.minHeight = minHeight
-        self.horizontalPadding = horizontalPadding
-        self.verticalPadding = verticalPadding
         self.content = content()
-    }
 
-    func itemWidth(forContentWidth contentWidth: CGFloat) -> CGFloat? {
         switch itemWidth {
-            case .ratio(let ratio):     return min(max(0, ratio * contentWidth - 2 * horizontalPadding), maxItemWidth ?? .infinity)
-            case .intrinsic:            return nil
-            case .custom(let width):    return min(width, maxItemWidth ?? .infinity)
+            case .intrinsic:                    self.snapping = false
+            case .ratio(_, let snapping):       self.snapping = snapping
+            case .custom(_, let snapping):      self.snapping = snapping
         }
     }
-}
 
-// MARK: - Previews
-struct HorizontalScrollPreviews: PreviewProvider {
+    private func scrollToID(id: AnyHashable?, geometry: GeometryProxy) {
+        guard let id else { return }
 
-    static var previews: some View {
-        PreviewWrapper {
-            content
+        if let itemToScrollTo = idPreferences.first(where: { $0.id == id }) {
```

Comment:
Could be combined with the `guard`.

Reply:
The implementation changed a bit, should be updated.